### PR TITLE
Completely test ethtool parsers for 1.x branch

### PR DESCRIPTION
**Pull request by PaulWay**
* Rewrote the main parsing of ring information for consistency
* Some better exception handling (not all)
* Tests now cover 100% of the code

### CODE REVIEW

**Comment by xiangce** [comment]
@PaulWay - It's ready to merge after you re-submit it to the master branch.

File: insights/parsers/ethtool.py [hunk 1]
```diff
@@ -816,31 +816,16 @@ def parse_content(self, content):
...
 
-        self._advertised_link_modes = []
+        self.advertised_link_modes = []
         if 'Advertised link modes' in self.data:
             for pair in self.data['Advertised link modes']:
-                self._advertised_link_modes += pair.split()
+                self.advertised_link_modes += pair.split()
 
-        self._supported_ports = []
+        self.supported_ports = []
```
Comment (xiangce) [inline insights/parsers/ethtool.py]: After doing that, you need to add these `Attributes` to the docstring explicitly.

**Comment by PaulWay** [comment]
Hi @xiangce - sorry for the delay, I was away.  The attributes in this are badly documented, so I'm going to document them throughout.

**Comment by PaulWay** [comment]
OK, @xiangce, I've updated this - let me know what you think now.

**Pull request merged**